<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Vehicle Monitor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <!-- Connection Panel -->
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>UART Connection</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>COM Port:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="cbComPort">
         <property name="minimumWidth">
          <number>120</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnRefresh">
         <property name="text">
          <string>Refresh</string>
         </property>
         <property name="maximumWidth">
          <number>80</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Baud Rate:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="cbBaudRate">
         <property name="minimumWidth">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnConnect">
         <property name="text">
          <string>Connect</string>
         </property>
         <property name="minimumWidth">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="lblStatus">
         <property name="text">
          <string>✗ Disconnected</string>
         </property>
         <property name="styleSheet">
          <string>color: red;</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>

    <item>
     <!-- Main Content -->
     <widget class="QWidget" name="widget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <!-- Left Panel: Tire Pressure -->
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Tire Pressure (PSI)</string>
         </property>
         <property name="minimumWidth">
          <number>300</number>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <!-- Front Left -->
            <item row="0" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Front Left:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLCDNumber" name="lcdTireFL">
              <property name="digitCount">
               <number>4</number>
              </property>
              <property name="segmentStyle">
               <enum>QLCDNumber::Flat</enum>
              </property>
             </widget>
            </item>

            <!-- Front Right -->
            <item row="1" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Front Right:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLCDNumber" name="lcdTireFR">
              <property name="digitCount">
               <number>4</number>
              </property>
              <property name="segmentStyle">
               <enum>QLCDNumber::Flat</enum>
              </property>
             </widget>
            </item>

            <!-- Rear Left -->
            <item row="2" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Rear Left:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLCDNumber" name="lcdTireRL">
              <property name="digitCount">
               <number>4</number>
              </property>
              <property name="segmentStyle">
               <enum>QLCDNumber::Flat</enum>
              </property>
             </widget>
            </item>

            <!-- Rear Right -->
            <item row="3" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Rear Right:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLCDNumber" name="lcdTireRR">
              <property name="digitCount">
               <number>4</number>
              </property>
              <property name="segmentStyle">
               <enum>QLCDNumber::Flat</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="lblTireWarning">
            <property name="text">
             <string>Status: OK</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>

          <!-- Engine Temperature Section -->
          <item>
           <widget class="QGroupBox" name="groupBox_4">
            <property name="title">
             <string>Engine Temperature</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QProgressBar" name="progressEngineTemp">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>120</number>
               </property>
               <property name="value">
                <number>90</number>
               </property>
               <property name="textVisible">
                <bool>true</bool>
               </property>
               <property name="format">
                <string>%v°C</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblEngineTempValue">
               <property name="text">
                <string>90.0 °C</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="styleSheet">
                <string>font-size: 18pt; font-weight: bold;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblEngineTempWarning">
               <property name="text">
                <string>⚠ WARNING</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>

       <!-- Right Panel: Battery Voltage -->
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Battery Voltage Monitor</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="lblBatteryVoltage">
            <property name="text">
             <string>12.6 V</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="styleSheet">
             <string>font-size: 24pt; font-weight: bold; color: #05B8CC;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblBatteryStatus">
            <property name="text">
             <string>Status: OK</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QChartView" name="chartView">
            <property name="minimumHeight">
             <number>300</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header>QtCharts</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
